arro3-io/src/avro/schema_store.rs

```diff
@@ -0,0 +1,182 @@
+use pyo3::exceptions::{PyTypeError, PyValueError};
+use pyo3::prelude::*;
+
+#[cfg(feature = "avro")]
+use arrow_avro::schema::{
+    AvroSchema, Fingerprint, FingerprintAlgorithm, SchemaStore as AvroSchemaStore,
+};
+
+#[cfg(feature = "avro")]
+#[derive(Clone, Copy)]
+pub(crate) enum AlgorithmType {
+    Rabin,
+    Id,
+    Id64,
+}
+
+#[cfg(feature = "avro")]
+impl From<AlgorithmType> for FingerprintAlgorithm {
+    fn from(val: AlgorithmType) -> Self {
+        match val {
+            AlgorithmType::Rabin => FingerprintAlgorithm::Rabin,
+            AlgorithmType::Id => FingerprintAlgorithm::Id,
+            AlgorithmType::Id64 => FingerprintAlgorithm::Id64,
+        }
+    }
+}
+
+/// Python wrapper for arrow-avro's SchemaStore
+///
+/// SchemaStore manages Avro writer schemas keyed by fingerprint.
+/// Each instance is configured with a FingerprintAlgorithm that determines
+/// how fingerprints are computed.
+#[cfg(feature = "avro")]
+#[pyclass(module = "arro3.io._io", name = "SchemaStore")]
+pub struct PySchemaStore {
+    inner: AvroSchemaStore,
+    algorithm: AlgorithmType,
+}
+
+#[cfg(feature = "avro")]
+#[pymethods]
+impl PySchemaStore {
+    /// Create a SchemaStore using Confluent fingerprints
+    #[staticmethod]
+    pub fn confluent() -> Self {
+        Self {
+            inner: AvroSchemaStore::new_with_type(FingerprintAlgorithm::Id),
+            algorithm: AlgorithmType::Id,
+        }
+    }
```

Comment:
How well known are these fingerprint types? Are these specific backend providers or generic to the spec?

I think it could be better if the SchemaStore had a generic constructor that allowed the user to pass in the fingerprint algorithm and algorithm type, rather than hard-coding specific provider types.